snuba/admin/views.py

```diff
@@ -81,20 +83,35 @@ def health() -> Response:
 
 @application.route("/migrations/groups")
 def migrations_groups() -> Response:
-    res: List[Mapping[str, MigrationGroup | Sequence[str]]] = []
-    for migration_group in get_active_migration_groups():
-        if migration_group.value in settings.ADMIN_ALLOWED_MIGRATION_GROUPS:
-            group_migrations = get_group_loader(migration_group).get_migrations()
-            res.append(
-                {"group": migration_group.value, "migration_ids": group_migrations}
-            )
+    res: List[Mapping[str, str | Sequence[Mapping[str, str | bool]]]] = []
+    allowed_groups = [
+        group.value
+        for group in get_active_migration_groups()
+        if group.value in settings.ADMIN_ALLOWED_MIGRATION_GROUPS
+    ]
+    if not allowed_groups:
+        return make_response(jsonify(res), 200)
+
+    for group, migrations in runner.show_all(allowed_groups):
+        migration_ids: Sequence[Mapping[str, str | bool]] = [
+            {
+                "migration_id": m.migration_id,
+                "status": m.status,
+                "blocking": m.blocking,
+                "can_run": m.can_run,
+                "can_reverse": m.can_reverse,
```

Comment:
Just to confirm, giving details about the reason a migration can/cannot be run to the UI is good (so it can be displayed to the user). Though are you also doing enforcing the check in the API method that actually executes/reverses the migration? That is mandatory to ensure nobody can go around the UI running migrations they do not have permissions to run.

Reply:
@fpacifici the `@check_migration_perms` on the run/reverse endpoints checks for the policies and the runner already has checks in place that are the same as the `StatusChecker`